### PR TITLE
added emails sending

### PR DESCRIPTION
**Pull request by iverhun**

### CODE REVIEW

File: src/main/java/com/lohika/morning/ecs/service/PublishCampaignService.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  private final EmailService emailService;
+  private final ExchangeService exchangeService;
+  private final EmailRecipientProvider emailRecipientProvider;
+
+  @Value("${email-client-service.recipient-email-override}")
+  private String recipientEmailOverride;
+
+  @Scheduled(cron = "${email-client-service.campaign-processing-schedule}")
+  public void publishAll() {
+    campaignService.getCampains(Campaign.Status.READY_TO_SEND).forEach(campaign -> {
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/service/PublishCampaignService.java]: fix job overlap

File: src/main/java/com/lohika/morning/ecs/domain/email/Email.java [hunk 2]
```diff
@@ -35,17 +36,23 @@
   @NonNull
   private String subject;
 
-  @Column(nullable = false)
+  @Column(nullable = false, length = 2_000)
```
[the remark this answers is not in the document]
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/domain/email/Email.java]: done

File: src/main/java/com/lohika/morning/ecs/domain/campaign/CampaignPreviewView.java [hunk 3]
```diff
@@ -68,6 +69,8 @@ public void enter(ViewChangeListener.ViewChangeEvent viewChangeEvent) {
     Campaign campaign = getCampaign(viewChangeEvent);
     binder.setBean(campaign);
     binder.bindInstanceFields(this);
+
+    publishButton.setEnabled(Campaign.Status.NEW == campaign.getStatus());
```
[the remark this answers is not in the document]
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/domain/campaign/CampaignPreviewView.java]: done

File: src/main/java/com/lohika/morning/ecs/domain/campaign/CampaignPreviewView.java [hunk 4]
```diff
@@ -32,34 +29,38 @@
...
+    cancelButton.addClickListener(this::cancelCampaign);
+    publishButton.addClickListener(this::startCampaign);
   }
 
-  private void generateEmails(Button.ClickEvent clickEvent) {
-    emailService.compileEmails(binder.getBean().getId());
+  private void startCampaign(Button.ClickEvent clickEvent) {
+    Campaign campaign = binder.getBean();
+    campaignService.updateStatus(campaign, Campaign.Status.PENDING);
+    publishButton.setEnabled(false);
```
[the remark this answers is not in the document]
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/domain/campaign/CampaignPreviewView.java]: done

File: src/main/java/com/lohika/morning/ecs/domain/email/EmailService.java [hunk 5]
```diff
@@ -27,22 +31,34 @@
...
     if (campaign.isAttendee()) {
-      // send to all
-      emails = attendeeService.findAll().stream().map(Attendee::getEmail).collect(toList());
+      // publish to all
+      emailAdresses = attendeeService.findAll().stream().map(Attendee::getEmail).collect(toList());
     } else {
-      // send to
-      emails = Arrays.asList(campaign.getEmails().split(","));
+      // publish to
+      emailAdresses = Arrays.asList(campaign.getEmails().split(","));
```
[the remark this answers is not in the document]
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/domain/email/EmailService.java]: used only here

File: src/main/java/com/lohika/morning/ecs/service/PublishCampaignService.java (hunk 1 above)
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/service/PublishCampaignService.java]: done

**Pull request merged**